lib/foreman_maintain/concerns/system_helpers.rb

```diff
@@ -92,6 +92,14 @@ def execute(command, options = {})
       def shellescape(string)
         Shellwords.escape(string)
       end
+
+      def fetch_etc_hostname
+        execute('hostname')
+      end
+
+      def fetch_server_ip
+        execute("ip route get 8.8.8.8 | awk 'NR==1 {print $NF}'").strip
```

Comment:
`ip route get <address>` command to ask the kernel to report the route it would use to send a packet to the specified address. 
Internet is not required for this as I have checked it by removing internet access and by clearing cache. It is still show same result even when I added invalid gateway.
I found that `ip route get`  is easiest way to fetch ip address without any complex logic.